ucf-degree-cpt.php

```diff
@@ -10,20 +10,41 @@
 	die;
 }
 
+define( 'UCF_DEGREE__PLUGIN_URL', plugins_url( basename( dirname( __FILE__ ) ) ) );
+define( 'UCF_DEGREE__PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
+define( 'UCF_DEGREE__STATIC_URL', UCF_DEGREE__PLUGIN_URL . '/static' );
+
+include_once 'includes/ucf-degree-program-type-tax.php';
+include_once 'includes/ucf-degree-career-path-tax.php';
+include_once 'includes/ucf-degree-posttype.php';
+include_once 'includes/ucf-degree-api.php';
+include_once 'includes/ucf-degree-utils.php';
+include_once 'admin/ucf-degree-admin.php';
+include_once 'admin/ucf-degree-config.php';
+
+if ( ! function_exists( 'ucf_degree_plugin_activation' ) ) {
+	function ucf_events_plugin_activation() {
+		UCF_Degree_Config::add_options();
+		flush_rewrite_rules();
+	}
+}
+
+if ( ! function_exists( 'ucf_degree_plugin_deactivation' ) ) {
+	function ucf_events_plugin_deactivation() {
+		UCF_Degree_Config::delete_options();
+	}
+}
+
 add_action( 'plugins_loaded', function() {
-	define( 'UCF_DEGREE__PLUGIN_URL', plugins_url( basename( dirname( __FILE__ ) ) ) );
-	define( 'UCF_DEGREE__PLUGIN_DIR', plugin_dir_path( __FILE__ ) );
-	define( 'UCF_DEGREE__STATIC_URL', UCF_DEGREE__PLUGIN_URL . '/static' );
-
-	include_once 'includes/ucf-degree-program-type-tax.php';
-	include_once 'includes/ucf-degree-career-path-tax.php';
-	include_once 'includes/ucf-degree-posttype.php';
-	include_once 'includes/ucf-degree-utils.php';
-	include_once 'admin/ucf-degree-admin.php';
-
-	add_action( 'init', array( 'UCF_Degree_ProgramType', 'register_programtype'), 0 );
-	add_action( 'init', array( 'UCF_Degree_CareerPath', 'register_careerpath' ), 0 );
-	add_action( 'init', array( 'UCF_Degree_PostType', 'register_degree_posttype' ), 0 );
+
+	add_action( 'init', array( 'UCF_Degree_ProgramType', 'register_programtype'), 10, 0 );
+	add_action( 'init', array( 'UCF_Degree_CareerPath', 'register_careerpath' ), 10, 0 );
+	add_action( 'init', array( 'UCF_Degree_PostType', 'register_degree_posttype' ), 10, 0 );
+
+	if ( is_plugin_active( 'rest-api/plugin.php' ) && UCF_Degree_Config::get_option_or_default( 'rest_api' ) ) {
```

Comment:
You could potentially use UCF_Degree_Config::rest_api_enabled() here instead of is_plugin_active() if you wanted to.